domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -497,6 +503,27 @@ private void init() throws Exception {
     }
 
     client = createPgPool(vertx, postgreSQLClientConfig);
+
+
+    vertx.setPeriodic(CHECK_FOR_QUERY_TIMEOUT_INTERVAL, PostgresClient::checkStaleConnections);
+
+  }
+
+  private static void checkStaleConnections(Long timerId) {
```

Comment:
```suggestion
  private static void checkStaleRequests(Long timerId) {
```
We want to cancel requests. Releasing connections is a different issue handled elsewhere (connection release delay).